packages/celotool/src/e2e-tests/transfer_tests.ts

```diff
@@ -414,12 +448,41 @@ describe('Transfer tests', function(this: any) {
     return { ok, fees, gas, events }
   }
 
+  function testTxPoolFiltering({
+    feeToken,
+    gas,
+    expectedError,
+  }: {
+    feeToken: CeloToken
+    gas: number
+    expectedError: string
+  }) {
+    it('should not add the transaction to the pool', async () => {
+      const feeCurrency =
+        feeToken === CeloContract.StableToken
+          ? await kit.registry.addressFor(CeloContract.StableToken)
+          : undefined
+      try {
+        const res = await transferCeloGold(FromAddress, ToAddress, TransferAmount, {
+          gas,
+          feeCurrency,
+        })
+        await res.waitReceipt()
+        assert.fail('no error was thrown')
+      } catch (error) {
+        assert.include(error.toString(), expectedError)
+      }
```

Comment:
Have you tried `test-utils.assertRevert`? Should cover this logic. Lots of examples in `accounts.ts` for reference

Reply:
That looks like it's for expecting VM reverts caused by failed require statements etc. whereas this is trying to expect an error that originates within geth